### PR TITLE
docs: add design document for HIP-904 maxAutoAssociations changes

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+3. Senders should cover the initial fees related to auto association
+4. Initialize auto- and lazy-created accounts to have unlimited number of auto associations
+
+## Non-Goals
+
+- Updating existing accounts’ `maxAutoAssociations` values
+
+## Architecture
+
+The implementation related to the new `maxAutoAssociations` behaviour will be gated behind a feature flag.
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md:26]: Let's give the feature flag a name here

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+We can reduce this friction by further expanding on the functionality around auto associations by having the ability to have unlimited number of auto association slots and removing the requirement for an account to prepay for unused token association slots. Instead we can have the sender pay for the initial number of auto association slots needed for a transfer.
+
+Additionally, in order to support behaviour similar to other EVM-compatible chains, we will want to initialize auto- and lazy-created accounts to have unlimited number of auto association slots.
+
+In order to implement [HIP-904](https://hips.hedera.com/hip/hip-904) we need to adapt part of the existing Services logic, so that we are aligned with the new approach.
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md:11]: Can we explicitly call out interfaces or classes in that will need to be updated in the sections below.
This way we can begin to mentally map and anticipate where changes will go

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+Contract accounts created by a contract create `EthereumTransaction` should have their `maxAutoAssociations` value default to 0.
+
+**EVM creations (`CREATE` & `CREATE2` opcodes)**
+
+Contract accounts by default will be spam-free in nature i.e. with `maxAutoAssociations` value default to 0.
+
+**`CREATE2` merge into existing hollow account**
+
+The current Services behaviour for this case is that `maxAutoAssociations` will be limited (positive integer) if the hollow account was created with token transfer or no `maxAutoAssociations` (0) if the hollow account was created with HBAR transfer.
```
Comment (Nana-EC) [inline hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md:64]: Can me more explicit here and say the 
```suggestion
The current Services behaviour for this case is that `maxAutoAssociations` will be limited to the number of already associated tokens in balance if the hollow account was created with token transfer or no `maxAutoAssociations` (0) if the hollow account was created with HBAR transfer.
```

**Review by Nana-EC — COMMENTED** [review]
Nice job capturing the details of the changes.
If you could call out the code components such as interfaces or key classes where these changes will be found that would paint a more clear picture

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @MiroslavGatsanoga !

**Review by Neeharika-Sompalli — DISMISSED** [review]
Agree with Nana's comments. LGTM otherwise. Thanks @MiroslavGatsanoga

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md (hunk 1 above)
Comment (MiroslavGatsanoga) [reply hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md]: Added as `tokens.airdrops.maxAutoAssociations.enabled`, let me know if some different naming makes better sense.

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md (hunk 2 above)
Comment (MiroslavGatsanoga) [reply hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md:11]: Added.

File: hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md (hunk 1 above)
Comment (MiroslavGatsanoga) [reply hedera-node/docs/design/services/token-service/airdrops/max-auto-associations-changes.md]: Updated to `tokens.airdrops.unlimitedMaxAutoAssociations.enabled` as it seemed more clear.

**Pull request merged**